lib/model/binding.dart

```diff
@@ -164,11 +262,68 @@ class LiveZulipBinding extends ZulipBinding {
   /// Initialize the binding if necessary, and ensure it is a [LiveZulipBinding].
   static LiveZulipBinding ensureInitialized() {
     if (ZulipBinding._instance == null) {
-      LiveZulipBinding();
+      final binding = LiveZulipBinding();
+      binding._deviceInfo = binding._prefetchDeviceInfo();
+      binding._packageInfo = binding._prefetchPackageInfo();
     }
     return ZulipBinding.instance as LiveZulipBinding;
   }
 
+  // Stored user agent header, since it remains constant.
+  Map<String, String>? _userAgentHeader;
+
+  @override
+  Future<BaseDeviceInfo?> get deviceInfo => _deviceInfo;
+  late Future<BaseDeviceInfo?> _deviceInfo;
+
+  @override
+  BaseDeviceInfo? get maybeDeviceInfo => _maybeDeviceInfo;
+  BaseDeviceInfo? _maybeDeviceInfo;
+
+  @override
+  Future<PackageInfo?> get packageInfo => _packageInfo;
+  late Future<PackageInfo?> _packageInfo;
+
+  @override
+  PackageInfo? get maybePackageInfo => _maybePackageInfo;
+  PackageInfo? _maybePackageInfo;
+
+  Future<BaseDeviceInfo?> _prefetchDeviceInfo() async {
```

Comment:
nit: keep "device info" members next to "device info" members, and "package info" members next to "package info" members

More generally, [as the Flutter style guide puts it](https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#order-other-class-members-in-a-way-that-makes-sense):
> Fields should come before the methods that manipulate them, if they are specific to a particular group of methods.

where the key part of "before" is that they're *next to* those methods, though in particular they should come just before them.